datafusion/optimizer/src/simplify_expressions/utils.rs

```diff
@@ -341,3 +355,49 @@ pub fn distribute_negation(expr: Expr) -> Expr {
         _ => Expr::Negative(Box::new(expr)),
     }
 }
+
```

Comment:
Do we need this? We already have an `Expr::is_volatile()` API.

Reply:
I'll refactor that